formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -1,691 +1,84 @@
-     * > Four CBOR items (arrays, maps, byte strings, and text strings) can be encoded with an indefinite length
-     */
-    private fun isIndefinite(): Boolean {
-        val majorType = curByte and 0b111_00000
-        val value = curByte and 0b000_11111
-
-        return value == ADDITIONAL_INFORMATION_INDEFINITE_LENGTH &&
-            (majorType == HEADER_ARRAY || majorType == HEADER_MAP ||
-                majorType == HEADER_BYTE_STRING.toInt() || majorType == HEADER_STRING.toInt())
-    }
-
-    /**
-     * Determines the length of the CBOR item represented by [curByte]; length has specific meaning based on the type:
-     *
-     * | Major type          | Length represents number of... |
-     * |---------------------|--------------------------------|
-     * | 0. unsigned integer | bytes                          |
-     * | 1. negative integer | bytes                          |
-     * | 2. byte string      | bytes                          |
-     * | 3. string           | bytes                          |
-     * | 4. array            | data items (values)            |
-     * | 5. map              | sub-items (keys + values)      |
-     * | 6. tag              | bytes                          |
-     */
-    private fun elementLength(): Int {
-        val majorType = curByte and 0b111_00000
-        val additionalInformation = curByte and 0b000_11111
-
-        return when (majorType) {
-            HEADER_BYTE_STRING.toInt(), HEADER_STRING.toInt(), HEADER_ARRAY -> readNumber().toInt()
-            HEADER_MAP -> readNumber().toInt() * 2
-            else -> when (additionalInformation) {
-                24 -> 1
-                25 -> 2
-                26 -> 4
-                27 -> 8
-                else -> 0
-            }
-        }
-    }
-
-    /**
-     * Indefinite-length byte sequences contain an unknown number of fixed-length byte sequences (chunks).
-     *
-     * @return [ByteArray] containing all of the concatenated bytes found in the buffer.
-     */
-    private fun readIndefiniteLengthBytes(): ByteArray {
-        val byteStrings = mutableListOf<ByteArray>()
-        do {
-            byteStrings.add(readBytes())
-            readByte()
-        } while (!isEnd())
-        return byteStrings.flatten()
-    }
-}
+@OptIn(ExperimentalSerializationApi::class)
+internal fun SerialDescriptor.getValueTags(index: Int): ULongArray? = findAnnotation<ValueTags>(index)?.tags
 
 @OptIn(ExperimentalSerializationApi::class)
-private fun SerialDescriptor.getElementIndexOrThrow(name: String): Int {
-    val index = getElementIndex(name)
-    if (index == CompositeDecoder.UNKNOWN_NAME)
-        throw SerializationException("$serialName does not contain element with name '$name." +
-            " You can enable 'CborBuilder.ignoreUnknownKeys' property to ignore unknown keys")
-    return index
-}
+internal fun SerialDescriptor.getObjectTags(index: Int): ULongArray? = findAnnotation<ObjectTags>(index)?.tags
```

Comment:
This function is unused, I guess it is not necessary since `@ObjectTags` cannot be specified on properties

Reply:
fixed in 7c25ffeb13ee78580575aa0219749a0ae90c39a5